skippy-gradle-android/src/main/java/io/skippy/gradle/android/ProjectSettings.java

```diff
@@ -27,17 +35,62 @@ private ProjectSettings(List<File> classesDirs, SkippyPluginExtension skippyExte
     }
 
     static ProjectSettings from(Project project) {
+        var buildDir = project.getLayout().getBuildDirectory().getAsFile().get().toPath();
         var skippyExtension = project.getExtensions().getByType(SkippyPluginExtension.class);
+        var projectDir = project.getProjectDir().toPath();
+        var classesDirs = new ArrayList<File>();
 
-        // TODO: figure out how to set the properties below for Android based projects
-        List<File> classesDirs = null;
-        Path projectDir = null;
-        Path buildDir = null;
+        project.getPlugins().withId("com.android.application", plugin -> {
+            var androidExtension = project.getExtensions().getByType(AppExtension.class);
```

Comment:
In android we can have 2 types of gradle modules: `application` with `AppExtension` or `library` with `LibraryExtension`.

Our plugin right now is working only with application type because we have `"com.android.application"` and `AppExtension`.

We can fix it by using `BaseExtension` or if it is not enough we can have something like:
```java
        AppExtension appExtension = project.getExtensions().findByType(AppExtension.class);
        LibraryExtension libraryExtension = project.getExtensions().findByType(LibraryExtension.class);

        if (appExtension != null) {
            handleApplication(appExtension);
        } else if (libraryExtension != null) {
            handleLibrary(libraryExtension);
        }
```

Reply:
Do you know if it is possible derive the folders with class files off of the `BaseExtension` object?

Reply:
I am investigating it right now :)

Reply:
good news, seems like I found a way to get all the android paths without using AGP at all. 
Bad news is that we need to use kotlin android plugin to get all the kotlin `class` files dirs. I will push an update soon.

Anyway is a good progress since we don't need to hardcode any paths anymore